src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -157,18 +109,50 @@ protected override async Task<ExecutionResult> ExecuteRequestAsync(HttpContext c
         }
         return await _documentExecuter.ExecuteAsync(opts);
     }
+
+    /// <inheritdoc/>
+    protected override IOperationMessageProcessor CreateMessageProcessor(IWebSocketConnection webSocketConnection, string subProtocol)
+    {
+        var authService = webSocketConnection.HttpContext.RequestServices.GetService<IWebSocketAuthenticationService>();
+
+        if (subProtocol == WebSockets.GraphQLWs.SubscriptionServer.SubProtocol)
+        {
+            return new WebSockets.GraphQLWs.SubscriptionServer(
+                webSocketConnection,
+                Options.WebSockets,
+                Options,
+                _documentExecuter,
+                _serializer,
+                _serviceScopeFactory,
+                this,
+                authService);
+        }
+        else if (subProtocol == WebSockets.SubscriptionsTransportWs.SubscriptionServer.SubProtocol)
+        {
+            return new WebSockets.SubscriptionsTransportWs.SubscriptionServer(
+                webSocketConnection,
+                Options.WebSockets,
+                Options,
+                _documentExecuter,
+                _serializer,
+                _serviceScopeFactory,
+                this,
+                authService);
+        }
+
+        throw new ArgumentOutOfRangeException(nameof(subProtocol));
+    }
 }
 
 /// <summary>
 /// ASP.NET Core middleware for processing GraphQL requests. Handles both single and batch requests,
 /// and dispatches WebSocket requests to the registered <see cref="IWebSocketHandler"/>.
 /// </summary>
-public abstract class GraphQLHttpMiddleware
+public abstract class GraphQLHttpMiddleware : IUserContextBuilder
 {
     private readonly IGraphQLTextSerializer _serializer;
-    private readonly IEnumerable<IWebSocketHandler>? _webSocketHandlers;
```

Comment:
I see warnings about `IWebSocketHandler` in xml comments.

Reply:
Fixed